coreapis/ldap/connection_pool.py

```diff
@@ -165,6 +165,9 @@ def search(self, base_dn, search_filter, scope, attributes, size_limit=None):
         alive_servers = self.alive_servers()
         if not alive_servers:
             alive_servers = self.servers
+        if len(alive_servers) == 1:
+            s = alive_servers.pop()
+            alive_servers = [s, s]
```

Comment:
I'd say that this is at least worth a comment in the code, so that one can see what the intent behind the code is. E.g. something like:

> If the organization has only one LDAP server, try it twice before giving up. This ensures that we don't fail simply because the the server happened to close the connection just before we were about to query it.

It may also be a good idea to rename the `alive_servers`-variable in this servers to something like `try_servers`, to reflect its actual contents, but that is sort of a different matter.